output/cloud/expv2/flush.go

```diff
@@ -87,15 +105,20 @@ type metricSetBuilder struct {
 	// It supports the iterative process for appending
 	// the aggregated measurements for each time series.
 	seriesIndex map[metrics.TimeSeries]uint
+
+	// discardedLabels tracks the labels that have been discarded
+	// because they are not supported by the remote service.
+	discardedLabels map[string]struct{}
 }
 
 func newMetricSetBuilder(testRunID string, aggrPeriodSec uint32) metricSetBuilder {
 	builder := metricSetBuilder{
 		MetricSet: &pbcloud.MetricSet{},
 		// TODO: evaluate if removing the pointer from pbcloud.Metric
 		// is a better trade-off
-		metrics:     make(map[*metrics.Metric]*pbcloud.Metric),
-		seriesIndex: make(map[metrics.TimeSeries]uint),
+		metrics:         make(map[*metrics.Metric]*pbcloud.Metric),
+		seriesIndex:     make(map[metrics.TimeSeries]uint),
+		discardedLabels: make(map[string]struct{}),
```

Comment:
We will invoke this code with big frequency and we should hit this some 0.x percentage so better keep it nil most of the time and init only if we need.

Reply:
:heavy_check_mark: with the latest commit